cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/AuthConfig.java

```diff
@@ -25,21 +25,21 @@
 public class AuthConfig {
 
   private final AuthType type;
-  private final String tokenName;
+  private final String passwordName;
```

Comment:
this and username seem like they are tied specifically to `AuthType.PAT` and may not always be relevant for other auth types. Do we know the other auth types that we will be supporting and the properties they require?

Reply:
Currently we're using the `UsernamePasswordCredentialsProvider` from jgit, which supports
1. The normal `username + password`;
2. The OAuth2 `(username) + accessToken(fetch through OAuth flow)`
3. PAT `(username) + token`

Reply:
@arjan-bal fyi

Reply:
it sounds like this means it will be different for each AuthType. I think it would be better to have a different proto class for each type then, rather than trying to fit them all into the same set of properties. For example:

```
public class AuthConfig {
  private final AuthType type;
  private final PATConfig patConfig;
}
```

kind of like how the Dataproc Job API (https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs#Job) has a different field for each type of possible job.

Reply:
@albertshau Added `PatConfig`, PTAL